k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -4889,18 +4889,25 @@ public void setSubjectProtected(boolean pEpSubjectProtection) {
 
     private synchronized Message appendpEpSyncMessage(final Account account, final Message message) throws MessagingException{
         Message localMessage = null;
-        LocalStore localStore = account.getLocalStore();
-        LocalFolder localFolder = localStore.getFolder(account.getCurrentpEpSyncFolderName());
-        localFolder.open(Folder.OPEN_MODE_RW);
+        try {
+            LocalStore localStore = account.getLocalStore();
+            LocalFolder localFolder = localStore.getFolder(account.getCurrentpEpSyncFolderName());
+            localFolder.open(Folder.OPEN_MODE_RW);
 
-        // Save the message to the store.
-        localFolder.appendMessages(Collections.singletonList(message));
-        // Fetch the message back from the store.  This is the Message that's returned to the caller.
-        localMessage = localFolder.getMessage(message.getUid());
+            // Save the message to the store.
+            localFolder.appendMessages(Collections.singletonList(message));
+            // Fetch the message back from the store.  This is the Message that's returned to the caller.
+            localMessage = localFolder.getMessage(message.getUid());
 
-        PendingCommand command = PendingAppend.create(account.getCurrentpEpSyncFolderName(), localMessage.getUid());
-        queuePendingCommand(account, command);
-        processPendingCommands(account);
+            PendingCommand command = PendingAppend.create(account.getCurrentpEpSyncFolderName(), localMessage.getUid());
+            queuePendingCommand(account, command);
+            processPendingCommands(account);
+        } catch (Exception e) {
+            Timber.e(e, "Exception appending a sync message");
+            if (e instanceof MessagingException) {
```

Comment:
isn't this a bit dangerous? Before we were throwing MessagingException, but unexpeced exceptions such as NullPointer would still bubble up. This way you catch Exception and only if it's a MessagingException you re-throw, but if it's a Nullpointer you just fail silent

Reply:
ah ok I see the description, we are doing it intentionally to avoid crashes. Ok then it makes sense, but we need then to find out the root cause by capturing more logs.

Reply:
I thought that it would not make sense in any case having a crash for any NullPointerException etc. Still I can just catch the IllegalStateException which is the one that is thrown when a deleted account is accessed at this point.

Regarding the root cause, it is more or less like this:
There are references to a certain account being handled in the background. When the account is deleted, even if the data etc of the account is removed, the references are still there. Then we have some work still being done on those references. And since the account was already removed, there is some error when the account's database is  tried to access.
In the current implementation we could try some things like adding many 'if' clauses for a deleted account. But to be honest I think that i could still happen because it is most likely that we will not add enough 'if' clauses. Therefore catching the exception to avoid the crash seems like a solution.
Also because we are talking about removing an account, which in production is not likely to be done very often. Then let's avoid the crash.

Reply:
Ok clear thanks for the explanation, I approved now